File: SpringTILCode/src/main/java/com/example/springtilcode/domain/user/controller/AuthController.java
package com.example.springtilcode.domain.user.controller;

import com.example.springtilcode.domain.user.controller.dto.request.AuthRequest;
import com.example.springtilcode.domain.user.controller.dto.response.TokenResponse;
import com.example.springtilcode.domain.user.service.AuthService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

@RestController
@RequiredArgsConstructor
public class AuthController {
    private final AuthService authService;

    @PostMapping("/signup")
    public void signup(@Valid @RequestBody AuthRequest request){
        authService.signup(request.getEmail(), request.getPassword());
    }

    @PostMapping("/login")
    public TokenResponse login(@Valid @RequestBody AuthRequest request){
        return authService.login(request.getEmail(), request.getPassword());
    }
}

File: kotlin/src/main/kotlin/com/example/demo/domain/user/controller/AuthController.kt
package com.example.demo.domain.user.controller

import com.example.demo.domain.user.controller.dto.request.UserRequest
import com.example.demo.domain.user.controller.dto.response.TokenResponse
import com.example.demo.domain.user.service.AuthService
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class AuthController (
    private val authService: AuthService
){
    @PostMapping("/signup")
    fun signup(@Valid @RequestBody userRequest: UserRequest){
        authService.signup(userRequest.email, userRequest.password)
    }

    @PostMapping("/login")
    fun login(@Valid @RequestBody userRequest: UserRequest): TokenResponse{
        return authService.login(userRequest.email, userRequest.password)
    }
}